applications/nrf_desktop/README.rst

```diff
@@ -1652,6 +1652,64 @@ For example, the following line will start the upload of the new image to the de
+The communication protocols are implemented using plugins.
+The plugin associated with the DFU protocol realized through the :ref:`nrf_desktop_config_channel` is branded as ``nordic_hid``.
+
+Adding a new device
+-------------------
+
+The device specifies which protocol is used for the communication by adding the specific device information to the plugin metadata.
+For the ``nordic_hid`` plugin this file is located at ``https://github.com/fwupd/fwupd/tree/main/plugins/nordic-hid/nordic-hid.quirk``
+The example entry is presented below:
+
+.. code-block:: console
+
+   [HIDRAW\VEN_1915&DEV_52DE]
+   Plugin = nordic_hid
+   GType = FuNordicHidCfgChannel
+   NordicHidBootloader = B0
+
+The first line describes the method of communication (``HIDRAW`` selects communication using ``HID`` reports) as well as vendor and device identifiers.
+This tells the fwupd tool where it should look for the device and how it should be identified.
+The next two lines set the plugin being used by the device.
+The final line ``NordicHidBootloader`` selects which bootloader the device is running.
+The possible values are either ``B0`` or ``MCUBOOT``.
+This information about bootloader are checked during both the connection to the device and during the update image compatibility verification.
+If the device does not present information about underlying bootloader or the bootloader type is incompatible, the fwupd tool will fail the image update.
+If the bootloader information contained withing a manifest being part of the update image package differ from the information presented by the device, the update procedure will fail as well.
+
+.. note::
+   Since the ``nordic_hid`` plugin communicate with the device using the :ref:`nrf_desktop_config_channel`, it will not allow device update through the :ref:`nrf_desktop_bootloader_serial_dfu`.
+
+To add a new device a pull request must be opened to the ``fwupd`` repository with a new entry to the :file:`nordic-hid.quirk` file.
+
+LVFS and update image preparation
+---------------------------------
+
+The LVFS project host firmware images that can be downloaded by Linux machines and used by the fwupt tool for firmware update of compatible devices.
+A valid account is needed to upload a new firmware image to the site.
+Information on how to apply for an account can be found at ``https://lvfs.readthedocs.io/en/latest/apply.html``.
+
+The nRF Desktop application DFU image is delivered as zip package containing a manifest and one or more binary files used for update.
+To prepare the image file compatible with the LVFS the CAB file needs to be prepared.
+The CAB package should contain the DFU package generated by the |NCS|, that is :file:`dfu_application.zip`, plus metadata file with information used by the LVFS.
+For more information about the LVFS metadata, see ``https://lvfs.readthedocs.io/en/latest/metainfo.html``.
+
+When CAB file is ready it needs to be signed and uploaded to the LVFS service.
```

Comment:
s/When CAB file is ready/Once the CAB archive has been built/

Reply:
@hughsie hm... I thought the signing of the CAB is not necessary for uploading, since the signing is done on LVFS side.

Reply:
Yes, that's right. It makes it sounds like the opposite here. Maybe instead:

    Once the CAB archive has been built it can be uploaded to the LVFS where it is verified and signed.